src/Chainweb/BlockHeaderDB.hs

```diff
@@ -316,9 +316,12 @@ dbAddChecked db e = unlessM (casMember (_chainDbCas db) ek) dbAddCheckedInternal
 
     -- TODO: make this atomic (create boilerplate to combine queries for
     -- different tables)
-    add = do
-        casInsert (_chainDbCas db) (RankedBlockHeader e)
-        tableInsert (_chainDbRankTable db) (_blockHash e) (_blockHeight e)
+    add = updateBatch
+            [ RocksDbInsert (_chainDbCas db) (casKey rbh) rbh
+            , RocksDbInsert (_chainDbRankTable db) (_blockHash e) (_blockHeight e)
+            ]
+      where
+        rbh = RankedBlockHeader e
```

Comment:
`rbh` is only used in one place, so the `where` clause might not be necessary.

Reply:
Actually, it's used in two places.

Reply:
My mistake!